faiss/impl/PanoramaStats.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// -*- c++ -*-
+
+#include <faiss/impl/PanoramaStats.h>
+
+#include <cstring>
+
+namespace faiss {
+
+void PanoramaStats::reset() {
+    total_dims_scanned = 0;
```

Comment:
@alexanderguzhva I agree with your comment on my previous PR that `memset((void*)this, 0, sizeof(*this));` isn't the cleanest way to implement this function. Out of curiosity, why do [`IndexIVFStats`](https://github.com/facebookresearch/faiss/blob/e6510bd00478d563e380604374f41a27e8697ce6/faiss/IndexIVF.cpp#L1282-L1284) and [`IndexBinaryHashStats`](https://github.com/facebookresearch/faiss/blob/e5fee1015a90dd3667c0d1f7d866bbc2c023c475/faiss/IndexBinaryHash.cpp#L278-L280) implement `reset()` like that, is there a particular benefit in those cases?